### PR TITLE
Bump Spark 3.3.0-rc5

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

https://www.mail-archive.com/dev@spark.apache.org/msg29070.html

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cfmcgrady** [comment]
we should also update the file `.github/workflows/master.yml`:

https://github.com/apache/incubator-kyuubi/blob/aabc53ec550dca23922defe3072759859d369bb2/.github/workflows/master.yml#L66

File: pom.xml [hunk 1]
```diff
@@ -2002,7 +2002,7 @@
             <properties>
                 <spark.version>3.3.0</spark.version>
                 <spark.archive.name>spark-${spark.version}-bin-hadoop3.tgz</spark.archive.name>
-                <spark.archive.mirror>https://dist.apache.org/repos/dist/dev/spark/v3.3.0-rc3-bin/</spark.archive.mirror>
+                <spark.archive.mirror>https://dist.apache.org/repos/dist/dev/spark/v3.3.0-rc4-bin/</spark.archive.mirror>
```
Comment (cfmcgrady) [inline pom.xml:2005]: rc5?

Comment (pan3793) [reply pom.xml]: Oops, fixed

**Review by cfmcgrady — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**